src/pyfmi/fmi_util.pyx

```diff
@@ -205,14 +259,32 @@ cpdef prepare_data_info(np.ndarray[int, ndim=2] data_info, list sorted_vars, lis
         last_index = last_index + 1
         data_info[1,i] = last_index
 
-    data = np.append(model.time, np.concatenate(
-                                    (model.get_real(param_real),
-                                    model.get_integer(param_int).astype(float),
-                                    model.get_boolean(param_bool).astype(float),
-                                    np.array(diagnostics_param_values).astype(float)),
-                                    axis = 0
-                                )
-                    )
+    if is_fmi3:
+        data = np.append(
+            model.time,
+            np.concatenate(
+                (model.get_float64(param_float64),
+                 model.get_float64(param_float32),
```

Comment:
This would fail if we had actual float32 variables?

Reply:
Typo, has been fixed now